src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int stSize = 10;
+    private int realSize = 0;
+    private T[] items;
+
+    public ArrayList() {
+        items = (T[]) new Object[stSize];
+    }
+
+    private void resize() {
+        if (realSize >= stSize) {
+            stSize *= 1.5;
+            T[] newItems = (T[]) new Object[stSize];
+            System.arraycopy(items, 0, newItems, 0, realSize);
+            items = newItems;
+        }
+    }
+
+    private void testIndexOk(int index) {
+        if (index < 0 || index >= realSize) {
+            throw new ArrayListIndexOutOfBoundsException("index is invalid: " + index);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        resize();
```

Comment:
you need to have methods `checkSize()` and `resize()` separately, because the size doesn't change every time you call method `resize()`
checkSize() - should check do you need to resize;
resize() - should resize your capacity